Assets/SEE/Layout/NodeLayouts/IncrementalTreeMapLayout.cs

```diff
@@ -0,0 +1,354 @@
+        }
+
+        /// <summary>
+        /// Collection all nodes of <see cref="_oldLayout"/> that are parent to a node,
+        /// with same id as a node in <paramref name="nodes"/>. The result will contain null
+        /// if there is a root in the old layout with a equivalent node in <paramref name="nodes"/>
+        /// </summary>
+        /// <param name="nodes"></param>
+        /// <returns></returns>
+        private ICollection<ILayoutNode> ParentsInOldGraph(IEnumerable<Node> nodes)
+        {
+            Assert.IsNotNull(_oldLayout);
+            HashSet<ILayoutNode> parents = new();
+            foreach (var node in nodes)
+            {
+                if (_oldLayout._iLayoutNodeMap.TryGetValue(node.ID, out var oldNode))
+                {
+                    parents.Add(oldNode.Parent);
+                }
+            }
+
+            return parents;
+        }
+
+        /// <summary>
+        /// The number of the nodes, that are also in the last layout present.
+        /// </summary>
+        /// <param name="nodes">the nodes to look up</param>
+        /// <returns></returns>
+        private int NumberOfOccurrencesInOldGraph(IEnumerable<Node> nodes)
+        {
+            Assert.IsNotNull(_oldLayout);
+            return nodes.Sum(n => _oldLayout._iLayoutNodeMap.ContainsKey(n.ID) ? 1 : 0);
+        }
+
+        /// <summary>
+        /// Adds the result of layout calculation to <see cref="_layoutResult"/>.
+        /// Applies padding to the result.
+        /// </summary>
+        /// <param name="nodes">nodes with calculated layout</param>
+        private void AddToLayout(IEnumerable<Node> nodes)
+        {
+            foreach (Node node in nodes)
+            {
+                float absolutePadding = _settings.paddingMm / 1000;
+                var rectangle = node.Rectangle;
+                var layoutNode = _iLayoutNodeMap[node.ID];
+
+                if (rectangle.Width - absolutePadding <= 0 ||
+                    rectangle.Depth - absolutePadding <= 0)
+                {
+                    absolutePadding = 0;
+                }
+
+                var position = new Vector3(
+                    (float)(rectangle.X + rectangle.Width / 2.0d),
+                    groundLevel,
+                    (float)(rectangle.Z + rectangle.Depth / 2.0d));
+                var scale = new Vector3(
+                    (float)(rectangle.Width - absolutePadding),
+                    layoutNode.LocalScale.y,
+                    (float)(rectangle.Depth - absolutePadding));
+
+                _layoutResult[layoutNode] = new NodeTransform(position, scale);
+            }
+        }
+
+        public override Dictionary<ILayoutNode, NodeTransform> Layout
+        (ICollection<ILayoutNode> layoutNodes, ICollection<Edge> edges,
+            ICollection<SublayoutLayoutNode> sublayouts)
+        {
+            throw new NotImplementedException();
```

Comment:
A comment stating why this will not be implemented (or adding a TODO that it should) would be nice here.